### ISSUE TITLE
Fix false positives for functions inside of math functions in length-zero-no-unit

### ISSUE THREAD

**Issue by tomyo** [stylelint/stylelint#5204]
> Clearly describe the bug
The rule `length-zero-no-unit` is applying fixes inside level 4 math functions.

This test exercises exactly what it shouldn't happen:

https://github.com/tomyo/stylelint/blob/eb419ef754ac442f9f3b9f7164fccb938b99adb3/lib/rules/length-zero-no-unit/__tests__/index.js#L332

Why?
It converts this:

```css
padding: calc(var(--foo, 0px) + 10px);
```
Into this:

```css
padding: calc(var(--foo, 0) + 10px);
```
If `--foo` is not defined, the fallback value `0` is used, resulting in `calc(0 + 10px);` which is invalid.

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "length-zero-no-unit": true
  }
}
```

> Which version of stylelint are you using?

13.12.0

> What did you expect to happen?

No unit strips should happen inside level 4 math functions, just for safety. 

In the link to the test I gave, something like this is what I would expect instead:

```js
{
			description: 'ignore calc, and inner functions',
			code: 'padding: calc(var(--foo, 0in) + 10px) 0px;',
			fixed: 'padding: calc(var(--foo, 0in) + 10px) 0;',
}
```

**Comment by jeddy3** [comment]
@tomyo Thanks for the report and clear write-up of the problem.

> No unit strips should happen inside level 4 math functions, just for safety.

Yes, that sounds appropriate.

Now that your related pull request (https://github.com/stylelint/stylelint/pull/5203) is merged, I'll label this is ready to implement.
